andela/suyabay
# [#120546921] Create endpoint that retrieves the comments of a specified user

#### What does this PR do?
- Write an endpoint that retrieves the comments of a particular user
- Leverage on the endpoint to retrieve the comment base on the query passed
#### Description of Task to be completed?
- Create a route `/v1/users/:name/comments` that calls the method `getUserComment` in the `CommentController`
-  Create `getUserComment` method to implement the user comments retrieval
-  Create two methods `displayUserComment` and `displayUserCommentsByDate` that returns the data leveraging on the userTransformer including the CommentTransformer
-  Use the userTransformer  to transform the user result
-  Write tests that assert that;
-  The user does not exist if the userName passed is invalid for the user with status code 404.
-  The comments of the user are retrieved with status code 200.
#### How should this be manually tested?
- We need POSTMAN for this to be manually tested, POSTMAN is an extension plugin on google chrome.
-  To retrieve the comments of the user,  send a `GET` request with this endpoint  `http://suyabay.app/api/v1/users/{name}/comments`. If the {users} is invalid, you should see a JSON response `The user does not exist`, status code `404`. If the {users} is valid, but there is no comments made by the user, you should see a JSON response `Comment not available for this user`, status code `404`. If the {users} is valid and the user has  comments, you should get all the comments made by the user with status code`200`.
-  To retrieve all comments made  the user with specified date,  send a `GET` request with this endpoint  `suyabay.app/api/v1/users/dehmohlah/comments?fromDate=2016-05-10&toDate=2016-05-18`. If the {username} is valid, you should see a JSON response with the comments requested from the recent comment to the oldest commence, a status code of `200`. If you add a limit, to the query like so:       `suyabay.app/api/v1/users/dehmohlah/comments?fromDate=2016-05-10&toDate=2016-05-18&limi=2`. If the {episode-name} is valid, you should see a JSON response with the comments requested from the recent comment to the oldest commence with the limit of 2, status code of `200`.
#### Any background context you want to provide?

None
#### What are the relevant pivotal tracker stories?
- #120546921
#### Screenshots (if appropriate)

![screen shot 2016-06-01 at 3 08 08 pm 4](https://cloud.githubusercontent.com/assets/14159420/15712571/bc69ea02-280a-11e6-8b25-0aec9818e9f5.png)

![screen shot 2016-06-01 at 3 09 50 pm](https://cloud.githubusercontent.com/assets/14159420/15712623/fed29678-280a-11e6-8742-6e6482025151.png)
![screen shot 2016-06-01 at 3 10 12 pm](https://cloud.githubusercontent.com/assets/14159420/15712624/fed3f8a6-280a-11e6-8160-eba25f2e83b2.png)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
[@andela-oadebayo](https://github.com/andela-oadebayo)
